rust/rsc/src/rsc/blob_store_impls.rs

```diff
@@ -32,17 +32,25 @@ impl BlobStore for LocalBlobStore {
     async fn stream<'a>(
         &self,
         stream: BoxStream<'a, Result<Bytes, std::io::Error>>,
-    ) -> Result<String, std::io::Error> {
+    ) -> Result<(String, i64), std::io::Error> {
         let reader = StreamReader::new(stream);
         futures::pin_mut!(reader);
 
         let key = create_temp_filename();
         let path = std::path::Path::new(&self.root).join(key.clone());
         let mut file = BufWriter::new(File::create(path).await?);
 
-        tokio::io::copy(&mut reader, &mut file).await?;
+        let written = tokio::io::copy(&mut reader, &mut file).await?;
 
-        Ok(key)
+        let size = match i64::try_from(written) {
```

Comment:
Is there a reason to not align the database type with the file writing type and make the database store `u64`?
The signed-ness here also doesn't seem to make logical sense to me.

Reply:
Yeah, postgres doesn't natively support `u64`. I actually really wanted `u64`